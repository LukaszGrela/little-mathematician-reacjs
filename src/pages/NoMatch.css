.no-match-404 {
    padding-top: 50px;
}

.no-match-404 .title {
    line-height: 3em;
}
.no-match-404 .title .sign {
    font-size: 1.5em;
    vertical-align: top;
}
.no-match-404 .title .digit {
    font-size: 3em;
}

.no-match-404 .digit.d1 {
    color: #388E3C;
}

.no-match-404 .digit.d2 {
    color: #8BC34A;
}

.no-match-404 .digit.d3 {
    color: #CDDC39;
}

.no-match-404 .title .sign {
    color: #FFEB3B;
}