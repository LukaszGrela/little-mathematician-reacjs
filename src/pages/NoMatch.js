import React, { Component } from 'react';

class NoMatch extends Component {
    state = {  }
    render() {
        return (
            <div>
                <h3>No match for <code>{this.props.location.pathname}</code></h3>
            </div>
        );
    }
}

export default NoMatch;