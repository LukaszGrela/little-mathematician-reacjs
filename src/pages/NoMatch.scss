/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import 'sass/_variables.scss';

.no-match-404 {
    padding-top: 50px;
}

.no-match-404 .title {
    line-height: 3em;
}
.no-match-404 .title .sign {
    font-size: 1.5em;
    vertical-align: top;
}
.no-match-404 .title .digit {
    font-size: 3em;
}

.no-match-404 .digit.d1 {
    color: $accent-color-1;
}

.no-match-404 .digit.d2 {
    color: $accent-color-2;
}

.no-match-404 .digit.d3 {
    color: $accent-color-3;
}

.no-match-404 .title .sign {
    color: $accent-color-4;
}