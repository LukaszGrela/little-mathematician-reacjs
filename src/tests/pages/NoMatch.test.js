import React from 'react';
import { shallow } from 'enzyme';
import NoMatch from '../../pages/NoMatch';

test('Should render NoMatch page correctly', () => {
    const wrapper = shallow(<NoMatch />);

    expect(wrapper).toMatchSnapshot();
});