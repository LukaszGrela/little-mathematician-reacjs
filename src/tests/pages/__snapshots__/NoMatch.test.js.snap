// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NoMatch page correctly 1`] = `
<div
  className="no-match-404"
>
  <div
    className="title"
  >
    <span
      className="digit d1"
    >
      4
    </span>
    <span
      className="sign"
    >
       - 
    </span>
    <span
      className="digit d2"
    >
      0
    </span>
    <span
      className="sign"
    >
       = 
    </span>
    <span
      className="digit d3"
    >
      4
    </span>
  </div>
  <div
    className="content"
  >
    Sorry, we can't find result you were counting on...
  </div>
</div>
`;
